milestone_1/crop_yield/crop_yield.md

```diff
@@ -0,0 +1,297 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.15.2
+  kernelspec:
+    display_name: Python 3 (ipykernel)
+    language: python
+    name: python3
+---
+
+```python
+# Read Shape files (.shp) with geopandas
+# Get .shp files link for the correspondant area at
+# http://download.geofabrik.de/ and place it in shape_files
+import datetime
+import glob
+import io
+import math
+import os
+import zipfile
+from collections import OrderedDict
+
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+from osgeo import gdal
+from pymodis import downmodis
+from pyproj import Proj
+```
+
+```python
+url = "http://download.geofabrik.de/europe/netherlands/gelderland-latest-free.shp.zip"
+extraction = "shape_files"
+if not os.path.exists(extraction):
+    os.makedirs(extraction)
+
+response = requests.get(url)
+if response.status_code == 200:
+    zipfile = zipfile.ZipFile(io.BytesIO(response.content))
+    zipfile.extractall(extraction)
+    print("Shape files downloaded")
+    print("Shape files are in shape_files")
+```
+
+```python
+gdf = gpd.read_file("shape_files/gis_osm_natural_free_1.shp")
+farthest_west = gdf.geometry.bounds["minx"].idxmin()
+
+# Find the farthest east point
+farthest_east = gdf.geometry.bounds["maxx"].idxmax()
+farthest_north = gdf.geometry.bounds["maxy"].idxmax()
+farthest_south = gdf.geometry.bounds["miny"].idxmin()
+west_coordinates = gdf.loc[farthest_west, "geometry"].coords[0]
+east_coordinates = gdf.loc[farthest_east, "geometry"].coords[0]
+north_coordinates = gdf.loc[farthest_north, "geometry"].coords[0]
+south_coordinates = gdf.loc[farthest_south, "geometry"].coords[0]
+print("Farthest West Coordinates:", west_coordinates)
+print("Farthest East Coordinates:", east_coordinates)
+print("Farthest North Coordinates:", north_coordinates)
+print("Farthest South Coordinates:", south_coordinates)
+```
+
+```python
+##Read Files using geopandas
+print(gdf.head())
+```
+
+```python
+"""""" """
+Brief Intro: MODIS assumes the Earth is a sphere
+with a R = 6.3781e6 m (6371007.181 m) radius.
+The world is subdivided into 18 vertical tiles and 36 horizontal tiles.
+each tile has 2400 cells of 500m x 500m.
+"""
+# Constants
+CELLS = 2400
```

Comment:
did you write the code yourself, if not, make sure to always add a link to the original code.

Reply:
Reference added